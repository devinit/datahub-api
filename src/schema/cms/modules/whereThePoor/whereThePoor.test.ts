import {getWhereThePoorPageData} from '.';
import * as prettyFormat from 'pretty-format';

describe('Where The Poor module tests', () => {
    it('should return page data for Where The Poor', async () => {
        const pageData: DH.IPage[] = await getWhereThePoorPageData();
        expect(pageData.length).toBeGreaterThan(0);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 10000);
});
