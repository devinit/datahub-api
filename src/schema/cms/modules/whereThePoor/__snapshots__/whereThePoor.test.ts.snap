// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Where The Poor module tests should return page data for Where The Poor 1`] = `
"Array [
  Object {
    \\"id\\": \\"poverty-visualisation\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Between 1990 and 2010, extreme poverty was cut in half and Explore plausible poverty outcomes for 2030 based on models of consumption\\",
    \\"title\\": \\"Where are the poor and where will they be?\\",
  },
  Object {
    \\"id\\": \\"Source\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Development Initiatives based on OECD DAC data.Note that figures are rounded: precise data are available for download on the methodology page\\",
    \\"title\\": \\"Source\\",
  },
  Object {
    \\"id\\": \\"download-the-data\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"For more information, see the DI data hub methodology page\\",
    \\"title\\": \\"Download the data\\",
  },
  Object {
    \\"id\\": \\"poverty-visualisation\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Between 1990 and 2010, extreme poverty was cut in half and Explore plausible poverty outcomes for 2030 based on models of consumption\\",
    \\"title\\": \\"Where are the poor and where will they be?\\",
  },
  Object {
    \\"id\\": \\"about-this-visualization\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"This page allows you to explore three scenarios for future poverty levels both globally and by region up to 2030. The baseline scenario uses current forecasts for growth in consumption and current inequality levels. The best-case scenario shows what happens with more rapid growth that is shared more equally. By contrast, the worst-case scenario sees slower growth and increasing inequality. Forecasting future poverty levels is difficult, given the inherent uncertainties involved. Poverty modelling develops future scenarios to help inform policymaking. These models rely on simplifying assumptions, often focusing on economic growth and income distribution trends which determine how much poor people benefit from that growth.Data accessed from source: 20 August 2013.\\",
    \\"title\\": \\"About this visualization\\",
  },
  Object {
    \\"id\\": \\"scenario details\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The best and worst case scenarios are based on the extremes of two elements: For growth, a +/- 2 percentage point margin of error either side of baseline projections of consumption growth (which are driven by economic growth). This is in line with differences observed between past forecasts and actual outcomes. For inequality, an annual 0.25 percentage point change in the shares of national consumption between the poorest 40% and the richest 10%\\",
    \\"title\\": \\"Scenario details\\",
  },
  Object {
    \\"id\\": \\"about-the-data\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Projections based on Chandy, Ledlie and Penciakova, The Final Countdown: Prospects for Ending Extreme Poverty by 2030 (2013), using the same methodology applied to a dataset updated with 40 country surveys. All data for years after 2010 are forecasts\\",
    \\"title\\": \\"About the data\\",
  },
]"
`;
